docs/api/tutorials/adding-dataset-custom-properties.md

```diff
@@ -0,0 +1,68 @@
+# Adding or Replacing Custom Properties on Datasets
+
+## Why Would You Add or Replace Custom Properties to Datasets? 
+Adding custom properties to datasets can help to provide additional information about the data that is not readily available in the standard metadata fields. Custom properties can be used to describe specific attributes of the data, such as the units of measurement used, the date range covered, or the geographical region the data pertains to. This can be particularly helpful when working with large and complex datasets, where additional context can help to ensure that the data is being used correctly and effectively.
+
+DataHub models custom properties of a Dataset as a map of key-value pairs of strings.
+
+Custom properties can also be used to enable advanced search and discovery capabilities, by allowing users to filter and sort datasets based on specific attributes. This can help users to quickly find and access the data they need, without having to manually review large numbers of datasets.
+
+### When to Add or Replace
+
+There may be situations where it is more appropriate to replace existing custom properties on a dataset, rather than simply adding new ones.
+
+Adding and removing specific properties is supported through patch semantics while full replaces can be done using the standard upsert APIs.
+
+
+### Goal Of This Guide
+This guide will show you how to add or replace custom properties on dataset `fct_users_deleted`.
+
+
+## Prerequisites
+For this tutorial, you need to deploy DataHub Quickstart and ingest sample data. 
+For detailed steps, please refer to [Prepare Local DataHub Environment](/docs/api/tutorials/references/prepare-datahub.md).
+
+:::note
+Before adding custom properties, you need to ensure the target dataset is already present in your DataHub instance. 
+If you attempt to manipulate entities that do not exist, your operation will fail.
+In this guide, we will be using data from sample ingestion.
+:::
+
+In this example, we will add some custom properties `cluster_name` and `retention_time` to the dataset `fct_users_deleted`.
+
+## Add Custom Properties With Java SDK
+The following code adds custom properties `cluster_name` and `retention_time` to a dataset named `fct_users_deleted` without affecting existing properties.
+
+```java
+{{ inline /metadata-integration/java/examples/src/main/java/io/datahubproject/examples/DatasetCustomProperties.java }}
+```
+
+## Replace Custom Properties With Java SDK
+The following code replaces the current custom properties to a dataset named `fct_users_deleted` with `cluster_name` and `retention_time`. Note, since this
+is utilizing the upsert API this will do a full replace of the Dataset Properties aspect and will not retain other fields unless specified. To just modify custom properties
```

Comment:
Shouldn't we be implementing the `REPLACE` operation in JsonPatch for this? (https://jsonpatch.com/#replace)

Reply:
Replace is supported, but it's kind of complicated to work with. I prefer pointing people to doing delete -> add